### ISSUE TITLE
Add Compute Credit Balance

### ISSUE THREAD

**Issue by tanishq-dubey** [acm-uiuc/groot-credits-service#12]
The Compute (acm-uiuc/clearwaters-container) and GPU cluster (acm-uiuc/gpu-cluster) will be online soon and the amount of time each user gets must be managed by some service. Initial thought was given to having the clusters themselves manage the amount compute time _(mapping from credits to time is to be discussed)_. However, it is better to have all the users and their time centrally managed.

One necessary feature that is different from merch credits is the idea of credit regeneration. Compute time is not paid for by members and so should regenerate at a certain rate. The rate based regeneration is required so that users cannot abuse credit refresh dates (such as end of week credit resets).

This should just be another column in the database and the corresponding API calls

**Comment by abrandemuehl** [comment]
Is there a way we can prioritize/limit usage without having a hard credit limit for time? I assume (correct me if I'm wrong) the cluster is going to spend most of it's time idle, and the rest will be doing long jobs. 

I don't want to have to "save up" credits for big jobs if no one is going to be using it anyways.

**Comment by tanishq-dubey** [comment]
You are correct, I am assuming the cluster will spend most of its time idle, unless we move nightly/batch tasks from our current compute source to the cluster. We could prioritize jobs based on who submits them, for example, if they are submitted on behalf of a SIG or ACM itself, while we hard-cap users. 

I think there is some incentive in capping the maximum number of credits you can have as it would essentially force users to use credits for whatever purpose (I can totally see people running their MPs on the cluster just for the sake of running it there).

**Comment by abrandemuehl** [comment]
Sounds like a good idea
